src/backend/cdb/motion/ic_tcp.c

```diff
@@ -1462,9 +1463,9 @@ SetupTCPInterconnect(EState *estate)
 
 		/* Outgoing connections */
 		outgoing_count = 0;
-		n = sendingChunkTransportState ? sendingChunkTransportState->numConns : 0;
+		numConns = sendingChunkTransportState ? sendingChunkTransportState->numConns : 0;
```

Comment:
Can it really be `NULL`? `sendingChunkTransportState` is returned from:
https://github.com/arenadata/gpdb/blob/df6f5babee0643069a3f3bbe13f97be1d1c35746/src/backend/cdb/motion/ic_tcp.c#L1376
Where:
https://github.com/arenadata/gpdb/blob/df6f5babee0643069a3f3bbe13f97be1d1c35746/src/backend/cdb/motion/ic_tcp.c#L514-L529

Reply:
it is initialized conditionally
https://github.com/arenadata/gpdb/blob/df6f5babee0643069a3f3bbe13f97be1d1c35746/src/backend/cdb/motion/ic_tcp.c#L1375-L1376

Reply:
Is this part alright?
https://github.com/arenadata/gpdb/blob/df6f5babee0643069a3f3bbe13f97be1d1c35746/src/backend/cdb/motion/ic_tcp.c#L1383

Reply:
yes, expectedTotalOutgoing is set only when startOutgoingConnections